ingest-asset-reconciler/src/main/scala/uk/gov/nationalarchives/Lambda.scala

```diff
@@ -60,62 +60,73 @@ class Lambda extends LambdaRunner[Input, StateOutput, Config, Dependencies] {
       entity <- IO.fromOption(entitiesWithAssetName.headOption)(
         new Exception(s"No entity found using SourceId '${asset.name}'")
       )
-
-      urlsToIoRepresentations <- dependencies.entityClient.getUrlsToIoRepresentations(entity.ref, Some(Preservation))
-      contentObjects <- urlsToIoRepresentations.map { urlToIoRepresentation =>
-        val generationVersion = urlToIoRepresentation.reverse.takeWhile(_ != '/').toInt
-        dependencies.entityClient.getContentObjectsFromRepresentation(entity.ref, Preservation, generationVersion)
-      }.flatSequence
-
-      _ <- log("Content Objects, belonging to the representation, have been retrieved from API")
-
-      stateOutput <-
-        if (contentObjects.isEmpty)
-          IO.pure(
-            StateOutput(wasReconciled = false, s"There were no Content Objects returned for entity ref '${entity.ref}'")
-          )
-        else {
+      children <- childrenOfAsset(dependencies.dynamoDbClient, asset, config.dynamoTableName, config.dynamoGsiName)
+      _ <- IO.fromOption(children.headOption)(
+        new Exception(s"No children were found for ${input.assetId} from ${input.batchId}")
+      )
+      _ <- log(s"${children.length} children found for asset ${asset.id}")
+      groupedChildren = children.groupBy(_.representationType match {
```

Comment:
1. Shouldn't we separate the representation type groups also, something like

```
      groupedChildren =
        children.groupBy (
          child =>
            (child.representationType, child.representationSuffix) match {
              case (DynamoFormatters.PreservationRepresentationType, suffix) => (Preservation, suffix)
              case (DynamoFormatters.AccessRepresentationType, suffix)       => (Access, suffix)
            }
        )

      stateOutputs <- groupedChildren
        .map { case ((representationType, suffix), childrenForRepresentationType) =>
```

2. Maybe `groupedChildren` could be `childrenGroupedByRepType`?

Reply:
1. I've tried this and it's not giving a `RepresentationType` as the key type of the Map because it can't tell that the match for `(child.representationType, child.representationSuffix)` is exhaustive (I think)
Aside from that, I don't think it's needed, the existing code and this are equivalent. 
2. Yeah, ok.

Reply:
I'm getting errors trying this further down. I could fix them but it's a fair bit of extra code and a type signature of `Map[(EntityClient.RepresentationType, Int), List[FileDynamoTable]]` for not much benefit so I'll leave it I think.
I've changed the variable name.